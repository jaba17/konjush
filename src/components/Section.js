import React, { Fragment, Component} from 'react';
import styled from 'styled-components';

const Title = styled.div`
  font-size: 28px;
  font-family: 'Pacifico';
`

const Subtitle = styled.div`
  margin-bottom: 10px;
  font-size: 17px;
`

const Section = class extends Component {
  
  shouldComponentUpdate(){
    return false;
  }

  render(){
    return (
    <Fragment>
      <Title className='w-100'>
        <p>{ this.props.title }</p>
      </Title>

      <Subtitle className='w-100'>
        <p>{ this.props.subtitle }</p>
      </Subtitle>
    </Fragment>
  )
  }
};

export default Section;