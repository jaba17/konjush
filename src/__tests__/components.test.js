import React from 'react';
import { shallow } from 'enzyme';
import Section from './../components/Section';

describe('>>> <Section/> must display each section properly <<<', function() {

  it('<Section/> must display text', function() {
    const wrapper = shallow(<Section title="About" subtitle="mission & how to contribute" />);
    expect(wrapper.find('p').at(0).html()).toMatchSnapshot();
    expect(wrapper.find('p').at(1).html()).toMatchSnapshot();
    expect(wrapper.find('p').length).toBe(2)
  })
  
})