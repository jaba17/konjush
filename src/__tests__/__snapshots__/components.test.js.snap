// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section/> must display each section properly <Section/> must display text 1`] = `"<p>About</p>"`;

exports[`<Section/> must display each section properly <Section/> must display text 2`] = `"<p>mission &amp; how to contribute</p>"`;
